Project21/Project21/Palindrome.c

```diff
@@ -0,0 +1,154 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment4 ~ A training program for new joiners at Metamation, Batch - Sept 2024.
+// Copyright (c) Naveenkumar S @Metamation India.
+// ------------------------------------------------------------------
+// Palindrome.c
+// C program to Check whether string or an integer is Palindrome or not
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <ctype.h>
+
+#define MAX_VALUE 100
+#define MAX_CHAR 200
+
+/// <summary>Clear the buffer values </summary>
+void ClrBuffer (char option) {
+   for (;;) {
+      option = fgetc (stdin);
+      if (option == '\n' || option == EOF) break;
+   }
+}
+
+/// <summary>Function to print result</summary>
+void PrintResult (bool value) {
+   printf (value ? ("\nIt is a Palindrome\n\n") : ("\nIt is not a Palindrome\n\n"));
+}
+
+/// <summary>Clear the content on screen</summary>
+void ClrScreen () {
+#ifdef _WIN32
+   system ("cls");
+#endif 
+}
+
+/// <summary>Function to check the given number is palindrome or not </summary>
+bool NumberIsPalindrome (int num) {
+   int org = num;
+   long long int rev = 0; // used to store reverse number exceeding integer range
+   while (num != 0) {
+      rev = rev * 10 + num % 10;
+      num /= 10;
+   }
+   if (org < 0) {
+      printf ("Output: %-7lld", rev);
+      rev = -rev;
+   }
+   else printf ("Output: %-7lld", rev);
+   return (rev == org);
+}
+
+/// <summary>Function to check the given string is palindrome or not </summary>
+bool StringIsPalindrome (char* sentences) {
+   int i = 0, j = 0, length = 0;
+   length = (int)strlen (sentences);
+   char temp[MAX_CHAR];
+   while (i < length) {
+      if (isalnum (sentences[i]) != 0)
+         temp[j++] = sentences[i];
+      i++;
+   }
+   temp[j] = '\0';
+   if (j == 0) return false;
+   for (int n = 0; n < (j / 2); n++) {
+      if (toupper (temp[n]) != toupper (temp[j - n - 1])) return false;
+   }
+   return true;
+}
+
+/// <summary>Function to get integer input from user</summary>
+void IntegerInput (int option) {
+   errno = 0;
+   char input[MAX_VALUE];
+   char* endptr;
+   printf ("\nEnter an integer: ");
+   char* i = fgets (input, sizeof (input), stdin);
+   char* intResult = strchr (input, '\n');
+   int number = strtol (input, &endptr, 10);

```

Comment:
The method fails to take the last digit if it is '0', e.g.: Input-0987890 should produce out as palindrome, but this implementation fails to achieve it.